### ISSUE TITLE
Level from blocks not respected for content parts

### ISSUE THREAD

**Issue by cstettler** [asciidoctor/asciidoctorj#681]
When `ContentPart` instances are created for each `AbstractBlock` from the AsciiDoc AST in `JRubyAsciidoctor.toDocument()`, the level is not read from the actual block (i.e. `AbstractBlock.getLevel()`), but is hardcoded to `1` for top-level blocks and subsequently increased for child blocks.

As a consequence, if an AsciiDoc document uses `leveloffset` to adjust the block level, the level on the content parts is not correct:

```
:leveloffset: -1
== Title
```

leads to a level value of `1` on the content part, but it should be `0` (as on the corresponding block)

**Comment by mojavelinux** [comment]
Keep in mind that the ContentPart API is deprecated. At this point, I don't think updates to it are wise. If this is a problem in the official AST that is part of the 1.6 branch, then we need to address it.

Here are the details about the AST in 1.6: https://github.com/asciidoctor/asciidoctorj/blob/asciidoctorj-1.6.0/docs/integrator-guide.adoc#understanding-the-ast-classes

**Comment by robertpanzer** [comment]
I can add a test for the AST classes on the 1.6 branch.
I wonder if it would still make sense to fix it for the ContentPart API if we add a few tests for it to ensure that?

**Comment by robertpanzer** [comment]
This problem only seems to happen when levels are skipped or out of order, e.g.
```
= Doc

== Section

==== Subsubsection

Oh no!!! My Section is reported as level 2 instead of 3...
```

Leveloffsets alone are not a problem, because the parser already brings them into the correct hierarchy.

**Comment by robertpanzer** [comment]
I can fix this for AsciidoctorJ 1.6.0.
I am reluctant to change it for 1.5 as I can't tell how many users are relying on this way of counting.
(I was already unclear if there are users at all using this API...)
